pinot-segment-local/src/main/java/org/apache/pinot/segment/local/indexsegment/mutable/MutableSegmentImpl.java

```diff
@@ -582,6 +583,9 @@ private void addNewRow(int docId, GenericRow row) {
           case DOUBLE:
             forwardIndex.setDouble(docId, ((Number) value).doubleValue());
             break;
+          case BIG_DECIMAL:
```

Comment:
Let's not add this case. Currently we don't support aggregated metric on `BIG_DECIMAL` type (no support for overriding var-length mutable dictionary entries)